### ISSUE TITLE
latest release does not have G4Data files

### ISSUE THREAD

**Issue by andresailer** [key4hep/key4hep-spack#148]
(This felt like the best fitting repository to open the issue in?)

To reproduce, ddsim fails to run with Geant4 complaining about missing file
```
source /cvmfs/sw.hsf.org/key4hep/setup.sh
ddsim --compactFile $lcgeo_DIR/share/lcgeo/compact/CLIC/compact/CLIC_o3_v14/CLIC_o3_v14.xml -G -N 10 --gun.distribution=uniform -O my_edm4hep.root
```
or 
```
source /cvmfs/sw.hsf.org/key4hep/setup.sh
env | grep G4
G4LEVELGAMMADATA=/cvmfs/sw.hsf.org/share/data/g4photonevaporation-5.7-vontejgdceozwf5ymnc6wyarevch72kg/share/data/PhotonEvaporation5.7
G4LEDATA=/cvmfs/sw.hsf.org/share/data/g4emlow-7.13-r5b7ydx472oljpocyxiruqqnie2tkc3d/share/data/G4EMLOW7.13
G4NEUTRONHPDATA=/cvmfs/sw.hsf.org/share/data/g4ndl-4.6-zlys4znfgnnxr7fqrct2afjshxxsgfdw/share/data/G4NDL4.6
G4ENSDFSTATEDATA=/cvmfs/sw.hsf.org/share/data/g4ensdfstate-2.3-jrydtxs2klc2zkqwdwslurjjmfbuh7nz/share/data/G4ENSDFSTATE2.3
G4RADIOACTIVEDATA=/cvmfs/sw.hsf.org/share/data/g4radioactivedecay-5.6-wyhngfhkfpwxbeo4udz7zpiyr5rw3bbb/share/data/RadioactiveDecay5.6
G4ABLADATA=/cvmfs/sw.hsf.org/share/data/g4abla-3.1-2jxylbld7geymksmd4lgddttt3sq5jdg/share/data/G4ABLA3.1
G4PIIDATA=/cvmfs/sw.hsf.org/share/data/g4pii-1.3-zp2rq2nom6bevtohjiftfuhc27n7xd55/share/data/G4PII1.3
G4PARTICLEXSDATA=/cvmfs/sw.hsf.org/share/data/g4particlexs-3.1-sj23ktyy4n62tyo2vf2igm2avkblu65f/share/data/G4PARTICLEXS3.1
G4SAIDXSDATA=/cvmfs/sw.hsf.org/share/data/g4saiddata-2.0-77yzjy4zeieavelx3vnef3ztm6uxyu2f/share/data/G4SAIDDATA2.0
G4REALSURFACEDATA=/cvmfs/sw.hsf.org/share/data/g4realsurface-2.2-t2np7vfwhwc5k7szhfdkjeitvazji4e4/share/data/RealSurface2.2
G4INCLDATA=/cvmfs/sw.hsf.org/share/data/g4incl-1.0-qsbncfghs3lzujqpewl7k6mibob4v5qh/share/data/G4INCL1.0
```

The `/cvmfs/sw.hsf.org/share/` folder is empty. The packages also don't live in 
```
 ls -l /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/g4ensdf* -d
drwxrwsrwx 4 cvmfs cvmfs 33 Nov  6 10:38 /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/g4ensdfstate-2.2-u3ihhqb7buve6cpfjjsexds6hucxpewp
```
Version 2.2 but not 2.3 is there for g4ensdfstate for example.

**Comment by vvolkl** [comment]
Thanks for the report, this indeed concerns this repo. For the latest release (in preparation of debug / macos builds), I moved the geant data files to `share/data` and declared them as external packages - this is a workaround for spack having no concept of "architecture-independent" packages that do not need a compiler. I discussed this with the devs, and they have this on the roadmap (overhaul of the way compilers are handled, making it possible to declare them as dependencies). This will likely still take a while and the `external` workaround was suggested.

This works, but the downside is that the installation workflow becomes more complicated, requiring an extra step to install the geant-data files to the share dir when they are updated. In this case, I did it on the build machine but then forgot  to sync the changes in share to the publisher, which I just did manually - so this particular issue should be fixed already.
